src/test/javascript/spec/component/programming-exercise/programming-exercise-instruction-analysis.component.spec.ts

```diff
@@ -93,6 +99,7 @@ describe('ProgrammingExerciseInstructionInstructorAnalysis', () => {
             // Test cases are not ok according to the analysis.
             expect(testCaseOk).toBeNull();
             expect(testCaseIssues).not.toBeNull();
+            discardPeriodicTasks();
```

Comment:
Without this line, the test always fails with the message `2 periodic timer(s) still in the queue.`. Does anyone know if this is a good solution or how this can be solved otherwise?

Reply:
Have you tried `flush()`? 
https://stackoverflow.com/questions/58416969/angular-karma-unit-test-error-1-timers-still-in-the-queue

Reply:
Yes, it did not change anything (I called it at the end instead of this method, is this the correct place?)

Reply:
Maybe you can call the ngOnDestroy to unsubscribe from the analysisSubscription?

Reply:
I was playing around with the test a bit and manged to solve it by adding another tick()-call.